Update to Protocol 14.1.0

craigcook opened the pull request:
## Description
Updates Protocol to v14.1.0 with a bunch of small tweaks to a few recently-added components (namely Split and Logo/Wordmark). This also touches a bunch of pages to tidy up some of the temporary customizations we made in bedrock while the changes were being made upstream in Protocol.

## Testing
You'll need to run `yarn` or `make build` to update local dependencies.

On demo at: https://www-demo4.allizom.org

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/374690221

alexgibson commented:
Now that https://github.com/mozilla/protocol/issues/687 is fixed, I think you should be able to remove this file: https://github.com/mozilla/bedrock/blob/master/media/static-bundles.json#L1508

alexgibson reviewed (changes requested):
I took a walk through the effected pages and everything looks good. Just one scrolling issue on the `/firefox/more/` page. r+wc 🕺 

Needs a rebase also.

craigcook commented on bedrock/firefox/templates/firefox/more.html:
```diff
@@ -18,11 +18,12 @@
     include_highres_image=True,
-    block_class='privacy-promise-hero mzp-l-split-hide-media-on-sm-md',
-    theme_class='mzp-t-dark'
+    block_class='page-hero mzp-l-split-hide-media-on-sm-md mzp-t-split-nospace',
```
I ran into this same issue on /firefox/privacy/promise/ which uses the exact same hero, so I can do the same fixes here.

craigcook commented:
Rebased and updated. The /firefox page could probably use a more extensive overhaul but maybe we can at least do some  visual cleanup in the near future.
